packages/core/core/src/services/document-service/webhooks.ts

```diff
@@ -0,0 +1,82 @@
+import { curry } from 'lodash/fp';
+
+import { UID, Schema, Utils, Core, Modules } from '@strapi/types';
+import { sanitize } from '@strapi/utils';
+
+import { getDeepPopulate } from './utils/populate';
+
+const ALLOWED_WEBHOOK_EVENTS = {
+  ENTRY_CREATE: 'entry.create',
+  ENTRY_UPDATE: 'entry.update',
+  ENTRY_DELETE: 'entry.delete',
+  ENTRY_PUBLISH: 'entry.publish',
+  ENTRY_UNPUBLISH: 'entry.unpublish',
+  ENTRY_DRAFT_DISCARD: 'entry.draft-discard',
+};
+
+type WebhookEvent = Utils.Object.Values<typeof ALLOWED_WEBHOOK_EVENTS>;
+
+const sanitizeEntry = async (
+  model: Schema.ContentType<any> | Schema.Component<any>,
+  entry: Modules.Documents.AnyDocument
+) => {
+  return sanitize.sanitizers.defaultSanitizeOutput(
+    {
+      schema: model,
+      getModel(uid) {
+        return strapi.getModel(uid as UID.Schema);
+      },
+    },
+    entry
+  );
+};
+
+/**
+ * Registers the content events that will be emitted using the document service.
+ */
+const registerEntryWebhooks = (strapi: Core.Strapi) => {
+  Object.entries(ALLOWED_WEBHOOK_EVENTS).forEach(([key, value]) => {
+    strapi.get('webhookStore').addAllowedEvent(key, value);
+  });
+};
+
+/**
+ * Triggers a webhook event.
+ *
+ * It will populate the entry if it is not a delete event.
+ * So the webhook payload will contain the full entry.
+ */
+const emitWebhook = async (
+  uid: UID.Schema,
+  eventName: WebhookEvent,
+  entry: Modules.Documents.AnyDocument
+) => {
```

Comment:
What I find odd about this function is that it's not webhook specific. What it technically does is just emit an event, which affects webhooks as much as audit logs or any potential other listener.

Reply:
Do you think we should prevent any type of event being userd in this function? it could easily be an if case